azure/Guidelines.md

```diff
@@ -477,6 +477,210 @@ Overall, this is a very brittle design that leads to a poor developer experience
 
 ### Collections
 
+The standard REST pattern for accessing a collection of resources is with a _GET_ method on the collection URL.
+This operation is commonly called the "list operation" for the resource, since the primary content of the response
+is an array (list) of the resource type.  Query parameters on the list operation may be provided to control
+which resources are returned and in what order.
+
+Resource collections can often be arbitrarily large, increasing lookup time as well as the size of the responses being sent over the wire. Therefore, it is important that list operations implement pagination.
+
+:white_check_mark: **DO** provide a list operation for each resource type.
+
+:white_check_mark: **DO** structure the response to a list operation as an object with a top-level array property to contain the collection of resources.
+
+:ballot_box_with_check: **YOU SHOULD** use _value_ as the name of the top-level array property -- there are some allowable exceptions to this guidance.
+
+:white_check_mark: **DO** implement pagination of list operation responses unless there is no possibility of a collection exceeding a size appropriate for a single response.
+In particular, include a top-level property in the response named _nextLink_ that contains an opaque URL to return the next page of results when there are additional items in the collection.
+
+:no_entry: **DO NOT** include a _nextLink_ property in the response if there are no additional items in the collection.
+
+:white_check_mark: **DO** clearly document that resources may be skipped or duplicated across pages of a paginated collection unless the operation has made special provisions to prevent this.
+
+Clients MUST be resilient to collection data being either paged or nonpaged for any given request.
```

Comment:
This comment is somewhat unclear to me - it could be interpreted such that a client should expect either an array OR an object with a value + nextLink URL for any given operation...

What we are trying to say is that a client should always be prepared to follow the `nextLink` to retrieve additional results....

Reply:
Do we want to include that sort of guidance here or https://azure.github.io/azure-sdk/general_introduction.html?

Reply:
I borrowed this from the MS API Guidelines but I agree that it is out of place here.  I think we need someplace to collect this kind of guidance for clients.  But I will remove it from here.